File: app/src/main/java/com/rogerio/KillFly/SpritesObjs.java
package com.rogerio.KillFly;
/*
 * Autor: Rogerio C. Santos - rogerio@rogeriocs.com.br
 * http://www.rogeriocs.com.br
 */
import libs.com.rogerio.utils.Sprite;
import libs.com.rogerio.utils.Vector2;
import android.content.Context;
import android.graphics.Canvas;

public class SpritesObjs{

	protected Sprite mSprite;
	protected Context mContext;

	public Sprite getSprite() {
		return mSprite;
	}

	public SpritesObjs() {
		
	}

	public void update(long time) {
		mSprite.Update(time);
	}

	public void draw(Canvas canvas) {
		mSprite.draw(canvas);
	}

	public void setPosition(Vector2 pos) {
		mSprite.setPosition(pos);
	}

	public Vector2 getPosition() {
		return mSprite.getPosition();
	}
	
	public void setRotate(float rot){
		mSprite.setRotate(rot);
	}

	public float getRotate(){
		return mSprite.getRotate();
	}
}

File: app/src/main/java/com/rogerio/KillFly/SpritesObjs.kt
package com.rogerio.KillFly

/*
 * Autor: Rogerio C. Santos - rogerio@rogeriocs.com.br
 * http://www.rogeriocs.com.br
 */
import libs.com.rogerio.utils.Sprite
import libs.com.rogerio.utils.Vector2
import android.content.Context
import android.graphics.Canvas

open class SpritesObjs {

    var sprite: Sprite? = null
        protected set
    protected var mContext: Context? = null

    var position: Vector2
        get() = sprite!!.position!!
        set(pos) {
            sprite!!.position = pos
        }

    var rotate: Float
        get() = sprite!!.rotate
        set(rot) {
            sprite!!.rotate = rot
        }

    fun update(time: Long) {
        sprite!!.Update(time)
    }

    fun draw(canvas: Canvas) {
        sprite!!.draw(canvas)
    }
}